app/src/components/helper/Save.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import { connect } from 'react-redux'
+import Check from '@material-ui/icons/Check'
+import CustomIconButton from './CustomIconButton'
+import { promises as fsPromise } from 'fs'
+import { SaveAlt } from '@material-ui/icons'
+import { bindActionCreators } from 'redux'
+import { rendererRpc } from '../../../../events'
+import { makeSaveDialogRpc } from '../../../../events/OpenDialogRequest'
+
+import { globalActions } from '../../actions'
+
+export async function saveToFile(buffer: Buffer | string): Promise<string | undefined> {
+  const rejectReasons = {
+    errorWritingFile: 'Error writing file',
+  }
+
+  const { canceled, filePath } = await rendererRpc.call(makeSaveDialogRpc(), {
+    securityScopedBookmarks: true,
+  })
+
+  if (!canceled && filePath !== undefined) {
+    try {
+      await fsPromise.writeFile(filePath, buffer)
```

Comment:
For some reason binary payloads do not get written with correct encoding

Reply:
The front-end should probably not directly write to the filesystem. 
This would block MQTT Explorer from becoming a web-app

Reply:
moved file operation to backend

Reply:
@thomasnordquist moved certificate selection file read to backend too. I have not tested it though since i do not use certs